### DISCUSSION TITLE
Is there a way to use Joi's API response to automatically populate input errors?

### DISCUSSION THREAD

**Discussion by mauriciosoares** [react-hook-form/react-hook-form#6916] [Q&A]
I'm using the `@hookform/resolvers/joi` resolver to use `Joi` to validate my form fields, everything works great.

However, I'm also using Joi on my api's (using https://github.com/arb/celebrate) to validate the api calls... whenever I send something to my API that is not valid, then my api will return something like this:

```
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "Validation failed",
  "validation": {
    "body": {
      "source": "body",
      "keys": ["email"],
      "message": "\"email\" must be a valid email"
    }
  }
}
```

I'm pretty sure that's a validation structure that Joi on my react app would understand and be able to parse, is there any way I can get this response and "inject" inside `joiResolver` so the errors from my API call would appear directly into my form fields?

**Comment by mauriciosoares** [comment]
No thoughts on this? If that's not even possible I'll probably work on a custom solution...

**Reply by bluebill1049** [reply]
i think you may need to build a custom resolver.
